### PR TITLE
obs-webkitgtk: init at unstable-2023-11-10

### PR DESCRIPTION
**Pull request by j-hui**
## Description of changes

Add `obs-webkitgtk` plugin, which is an alternative "Browser" source to the default `obs-browser` plugin (based on CEF).

To test this, I built OBS with this plugin using the following derivation:

```nix
{ pkgs ? import ../nixpkgs/default.nix {} }:
(pkgs.wrapOBS {
  plugins = [
    pkgs.obs-studio-plugins.obs-webkitgtk
  ];
})
```

Ran `nix-build` and `./results/bin/obs` to test.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by MaxCan-Code** [comment]
Result of `nixpkgs-review pr 284887` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>obs-studio-plugins.obs-webkitgtk</li>
  </ul>
</details>

**Review by fpletz — APPROVED** [review]

File: pkgs/applications/video/obs-studio/plugins/obs-webkitgtk.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+, meson
+, cmake
+, pkg-config
+, ninja
+, wrapGAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "obs-webkitgtk";
+  version = "ddf230852c3c338e69b248bdf453a0630f1298a7";
```
Comment (fpletz) [inline pkgs/applications/video/obs-studio/plugins/obs-webkitgtk.nix:16]:
```suggestion
  version = "unstable-2023-11-10";
```

We use the commit date prepended by unstable as version for unreleased versions.

File: pkgs/applications/video/obs-studio/plugins/obs-webkitgtk.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "obs-webkitgtk";
+  version = "unstable-2023-11-10";
+
+  src = fetchFromGitHub {
+    owner = "fzwoch";
+    repo = "obs-webkitgtk";
+    rev = version;
```
Comment (j-hui) [inline pkgs/applications/video/obs-studio/plugins/obs-webkitgtk.nix:21]: @fpletz should I still use the commit hash for the `fetchFromGitHub` invocation?

```suggestion
    rev = "ddf230852c3c338e69b248bdf453a0630f1298a7";
```

Comment (fpletz) [reply pkgs/applications/video/obs-studio/plugins/obs-webkitgtk.nix:21]: Oh, right! ;)

**Comment by fpletz** [comment]
Let's wait for the CI. I'm gonna squash the commits in the merge.

**Pull request merged**